src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -53,6 +55,11 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
             $this->tokenStorage->setToken($token = $this->authenticationManager->authenticate($token));
         }
 
-        return $this->accessDecisionManager->decide($token, [$attribute], $subject);
+        return ($this->lastAccessDecision = $this->accessDecisionManager->decide($token, [$attribute], $subject))->isGranted();
```

Comment:
I'd suggest to add a `trigger_error` if the `decide()` method returns a boolean, for the BC layer

Reply:
We have to trigger a deprecated in that place + in the implementations of `AccessDecisionManagerInterface` which contain the `decide()` method, right ?

Reply:
I added a second deprecation when the VoterInterface::vote() method returns an int.